src/Controls/tests/Core.UnitTests/PageTests.cs

```diff
@@ -579,5 +579,27 @@ public void LogicalChildrenDontAddToPagesInternalChildren()
 			Assert.Contains(customControl, page.LogicalChildrenInternal);
 			Assert.Contains(customControl, ((IVisualTreeElement)page).GetVisualChildren());
 		}
+
+		[Fact]
+		public void SendDisappearingUnsubscribesAppearingEvent()
+		{
+			var page = new ContentPage();
+			var navPage = new NavigationPage(page);
+			_ = new TestWindow(navPage);
+			
+			// This subscribes the event if the page has not appeared already
+			int invokeCounter = 0;
+			navPage.OnAppearing(() => { invokeCounter++; });
+			((IPageController)navPage).SendAppearing();
+			Assert.Equal(1, invokeCounter);
+			
+			// Navigate away
+			((IPageController)navPage).SendDisappearing();
+			
+			// This should not increment the counter when navigating back
+			navPage.OnAppearing(() => { invokeCounter++; });
+			((IPageController)navPage).SendAppearing();
+			Assert.Equal(1, invokeCounter);
```

Comment:
This may work better:

```suggestion
			// This subscribes the event if the page has not appeared already
			page.OnAppearing(() => Assert.Fail("Previous OnAppearing event code ran when it should have been removed."));

			// Navigate away BEFORE a navigation TO completes
			((IPageController)navPage).SendDisappearing();

			// This should not increment the counter when navigating back
			((IPageController)navPage).SendAppearing();
```

But it still does not work and I am going to bed. :)

But basically, you are calling SendAppearing which will always trigger and then unsub the event, so maybe you should skip the first SendAppearing to emulate? Or maybe this issue is a bit different? Might need to finish my nap before I fully understand this issue.

Reply:
yeah I was seeing something similar on my end too, will look into it today, thanks I like that Assert.Fail()